omni-relayer/src/workers/near.rs

```diff
@@ -0,0 +1,255 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use futures::future::join_all;
+use log::{error, info, warn};
+
+use alloy::rpc::types::Log;
+use ethereum_types::H256;
+use near_primitives::borsh::BorshSerialize;
+use nep141_connector::Nep141Connector;
+use omni_types::{
+    locker_args::ClaimFeeArgs, near_events::Nep141LockerEvent, prover_args::EvmVerifyProofArgs,
+    prover_result::ProofKind, ChainKind,
+};
+
+use crate::{config, utils};
+
+pub async fn sign_transfer(
+    config: config::Config,
+    redis_client: redis::Client,
+    connector: Arc<Nep141Connector>,
+) -> Result<()> {
+    let redis_connection = redis_client.get_multiplexed_tokio_connection().await?;
+
+    loop {
+        let mut redis_connection_clone = redis_connection.clone();
+        let Some(mut events) = utils::redis::get_events(
+            &mut redis_connection_clone,
+            config.redis.near_init_transfer_events.clone(),
+        )
+        .await
+        else {
+            tokio::time::sleep(tokio::time::Duration::from_secs(
+                config.redis.sleep_time_after_events_process_secs,
+            ))
+            .await;
+            continue;
+        };
+
+        let mut handlers = Vec::new();
+        while let Some((nonce, event)) = events.next_item().await {
+            if let Ok(event) = serde_json::from_str::<Nep141LockerEvent>(&event) {
+                handlers.push(tokio::spawn({
+                    let config = config.clone();
+                    let mut redis_connection = redis_connection.clone();
+                    let connector = connector.clone();
+
+                    async move {
+                        let Nep141LockerEvent::InitTransferEvent { transfer_message } = event
+                        else {
+                            warn!("Expected InitTransferEvent, got: {:?}", event);
+                            return;
+                        };
+
+                        info!(
+                            "Received InitTransferEvent: {}",
+                            transfer_message.origin_nonce.0
+                        );
+
+                        match connector
+                            .sign_transfer(
+                                transfer_message.origin_nonce.into(),
+                                Some(config.testnet.token_locker_id),
+                                transfer_message.fee.into(),
+                            )
```

Comment:
We probably want to provide our NEAR account here.

Reply:
We're providing it in startup/mod.rs build_connector function and `sign_transfer` just uses provided token_locker_id that is provided in connector

Reply:
This field is missing: https://github.com/Near-One/rainbow-bridge-sdk-rs/blob/omni-token-support/bridge-sdk/connectors/omni-connector/src/omni_connector.rs#L324

Reply:
Sorry, I still don't understand why it's missing. We are setting near signer here: https://github.com/Near-One/omni-bridge/blob/ed02e1475c33aab169d644523ed7a2fa5e4e4950/omni-relayer/src/startup/mod.rs#L28-L29

and then creating new `InMemorySigner` here:
https://github.com/Near-One/rainbow-bridge-sdk-rs/blob/omni-token-support/bridge-sdk/connectors/omni-connector/src/omni_connector.rs#L481-L495

Moreover, we have the same logic in `sign_transfer` and it works perfectly fine as we tested it today:
https://github.com/Near-One/rainbow-bridge-sdk-rs/blob/omni-token-support/bridge-sdk/connectors/omni-connector/src/omni_connector.rs#L341-L372